ARCHITECTURE.md

```diff
@@ -0,0 +1,40 @@
+Following diagram describes the interaction of a User with CLA CI Action.
+
+```mermaid
+sequenceDiagram
+    participant User as User
+    participant Core as Action Core
+    participant GitHub as GitHub
+    participant CLAService as CLA Web Service
+
+    User->>Core: Trigger action with implicit<br> GitHub token
+    Core->>GitHub: Retrieve commits from<br> the pull request
+
+    loop Process each commit
+        Core->>Core: Check commit message for a license
+        alt License valid
+            Core->>Core: Log license approval
+        else No valid license
+            Core->>Core: Collect author details
+        end
+    end
+
+    Core->>Core: Process CLA exceptions<br> (for bots and Canonical employees)<br> for authors
+
+    Core->>CLAService: Check CLA status for authors
+    CLAService->>CLAService: Validate against Individual CLA DB
+    CLAService->>CLAService: Check if author email domain<br> is in the corporate CLA DB
+    CLAService-->>Core: Return CLA validation results
+
+    loop For each commit author
+        Core->>Core: Report CLA status for author
+    end
+    
+    alt All authors signed CLA
+        Core->>Core: Log success message
+    else Some authors have not signed CLA
+        Core->>Core: Log failure message
+        Core->>User: Notify that action failed
```

Comment:
Is there a special notification there or just the fact that the action is failing? If it's just the result then for consistency it would make sense to have 'Core->>User: Notify that action succeeded' in the outcome is positive?

Reply:
I have referred that we set core failed in the code, indiciating non zero exit code, which sets action to failed

If we don't do anything, then action is automatically green, as exit code is 0, implicitly

From business logic perspective I can add it
From code perspective we don't do anything for positive action result

How would you prefer in diagram?

Reply:
I see the diagram value being more for users who want an overview of what the action is doing, from that perspective communicating the positive outcome as a return of the action to the user makes sense so yes please add it, thanks!

Reply:
done